source/model.js

```diff
@@ -37,52 +50,97 @@ let singletonField = (function(){
     };
 
     return {
-        getInstance: function(){
+        getInstance: function(rows, columns){
             if(!instance){
-                instance = createInstance();
+                instance = new Field(rows, columns);
             }
             return instance;
         }
     };
 })();
 
-function WinnerChecker(field){
-    let winCombinations = [
-        [0, 1, 2], [3, 4, 5], [6, 7, 8],
-        [0, 3, 6], [1, 4, 7], [2, 5, 8],
-        [0, 4, 8], [2, 4, 6]
+function WinnerChecker(field, winCombinationLength){
+    let directions = [
+        [0, 4], [1, 5], [2, 6], [3, 7]
     ];
 
-    this.getWinner = function (cellNum, symbol){
-        let winArray = [];
-        for (let i = 0; i < winCombinations.length; i++){
-            if (winCombinations[i].indexOf(cellNum) != -1) {
-                winArray.push(winCombinations[i]);
-            }
-        }
-        for (let winSegment in winArray) {
-            let isWinnerFound = true;
-            for (let i = 0; i < winArray[winSegment].length; i++) {
-                if (field.fieldCells[winArray[winSegment][i]] != symbol) {
-                    isWinnerFound = false;
-                    break;
+    this.getWinner = function(row, col, obj){
+        for (let i = 0; i < directions.length; i++) {
+            let winArray = [];
+            let currentCell = [{row: row, col: col, object: obj}];
+            winArray.push(currentCell);
+            for (let j = 0; j < directions[i].length; j++) {
+                let nextGameCell = getNextCell(currentCell, directions[i][j]);
+                while (nextGameCell !== null) {
+                    winArray.push(nextGameCell);
+                    if (winArray.length === winCombinationLength){
+                        return true;
+                    }
+                    nextGameCell = getNextCell(nextGameCell, directions[i][j]);
                 }
             }
-            if (isWinnerFound) {
-                return winArray[winSegment];
-            }
         }
         return false;
     };
+
+    function getNextCell(currentCell, direction){
+        let nextCell;
+        switch(direction){
+            case 0: /*up*/
+                nextCell = [{row: currentCell[0].row-1, col: currentCell[0].col, object: currentCell[0].object}];
+                break;
+            case 1: /*top right*/
```

Comment:
These directions' names depends on American or British English and are correct in both variants.